src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -24,19 +24,17 @@ internal class MsalServiceExceptionFactory
             MsalServiceException ex = null;
             var oAuth2Response = JsonHelper.TryToDeserializeFromJson<OAuth2ResponseBase>(httpResponse?.Body);
 
-            if (string.Equals(oAuth2Response?.Error, MsalError.InvalidGrantError, StringComparison.OrdinalIgnoreCase))
+            if (IsInvalidGrant(oAuth2Response) || IsInteractionRequired(oAuth2Response))
```

Comment:
Turns out that MSALs should also throw `UI Req Exception` if error_code is "interaction_required"

Reply:
makes sense.

---
In reply to: [420305866](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1804#discussion_r420305866) [](ancestors = 420305866)